src/scripting/KM_ScriptingUtils.pas

```diff
@@ -301,6 +303,20 @@ function TKMScriptUtils.EnsureRangeS(aValue, aMin, aMax: Single): Single;
 
 
 //* Version: 7000+
+//* Wrapper for pascal Format function
+//* Formats aFormatting string with specified aData array of parameters
+function TKMScriptUtils.Format(aFormatting: string; aData: array of const): string;
+begin
+  try

```

Comment:
Let's add validation here. Compare number of passed parameters and number of % occurring in a string.